NetKet/Graph/custom_graph.hpp

```diff
@@ -155,12 +284,41 @@ class CustomGraph : public AbstractGraph {
         }
       }
     }
+  }
+
+  /// Checks that for each edge (i, j): 0 <= i <= j and returns max(j) + 1, i.e.
+  /// the number of nodes
+  int CheckEdges() const {
```

Comment:
Comment on the fact that this also returns the number of sites

Reply:
@gcarleo you mean the reason for it? Quite simple really: if you're only given a list of edges how do you figure out how many nodes the graph contains? The answer is by going through the edges. So why do it twice when you can obtain the number of sites for free from `CheckEdges`?